### PR TITLE
feat: adds support for performing mass operations on issues

### PR DESCRIPTION
**Pull request by bcoe**
I wanted to be able to list any of our libraries that have an unclosed GA issue opened:

```
  https://github.com/googleapis/nodejs-billing/issues/3           GA Release of @google-cloud/billing
  https://github.com/googleapis/nodejs-dialogflow/issues/467      GA release
  https://github.com/googleapis/nodejs-document-ai/issues/11      promote library to GA
  https://github.com/googleapis/nodejs-media-translation/issues/3 promote library to GA
  https://github.com/googleapis/nodejs-memcache/issues/4          promote library to GA
  https://github.com/googleapis/nodejs-os-config/issues/2         promote library to GA
  https://github.com/googleapis/nodejs-recommender/issues/10      promoting library to GA
  https://github.com/googleapis/nodejs-service-directory/issues/4 Promote library to GA
```

This could be used to perform other mass operations on issues (I think someone asked for this in our weekly meeting today).

### CODE REVIEW

File: src/lib/asyncPrIterator.ts [hunk 1]
```diff
@@ -91,68 +111,120 @@ export async function process(
...
-    prs = prs.filter(prSet => prSet.pr.user.login === cli.flags.author);
+    items = items.filter(
+      itemSet => itemSet.item.user.login === cli.flags.author
+    );
   }
   orb1.succeed(
-    `[${scanned}/${repos.length}] repositories scanned, ${prs.length} matching PRs found`
+    `[${scanned}/${repos.length}] repositories scanned, ${
+      items.length
+    } matching ${processIssues ? 'issues' : 'PR'}s found`
```
Comment (alexander-fenster) [inline src/lib/asyncPrIterator.ts:136]: nit: extra `s` (will print `issuess`)

File: src/lib/asyncPrIterator.ts [hunk 2]
```diff
@@ -91,68 +111,120 @@ export async function process(
...
+    items.map(itemSet => {
       return async () => {
-        const title = prSet.pr.title!;
+        const title = itemSet.item.title!;
         if (title.match(regex)) {
-          orb2.text = `[${processed}/${prs.length}] ${options.commandActive} PRs`;
-          const result = await options.processMethod(prSet.repo, prSet.pr, cli);
+          orb2.text = `[${processed}/${items.length}] ${
+            options.commandActive
+          } ${processIssues ? 'issues' : 'PR'}s`;
```
Comment (alexander-fenster) [inline src/lib/asyncPrIterator.ts:150]: same here

File: src/lib/asyncPrIterator.ts [hunk 3]
```diff
@@ -91,68 +111,120 @@ export async function process(
...
+        item.html_url.length > maxLength ? item.html_url.length : maxLength,
       0
     );
 
-  console.log(`Successfully processed: ${successful.length} pull request(s)`);
-  for (const pr of successful) {
-    console.log(`  ${pr.html_url.padEnd(maxUrlLength, ' ')} ${pr.title}`);
+  console.log(
+    `Successfully processed: ${successful.length} ${
+      processIssues ? 'issues' : 'PR'
```
Comment (alexander-fenster) [inline src/lib/asyncPrIterator.ts:201]: same here

File: src/lib/asyncPrIterator.ts [hunk 4]
```diff
@@ -91,68 +111,120 @@ export async function process(
...
+    }s`
+  );
+  for (const item of successful) {
+    console.log(`  ${item.html_url.padEnd(maxUrlLength, ' ')} ${item.title}`);
   }
 
   if (failed.length > 0) {
     console.log(`Unable to process: ${failed.length} pull requests(s)`);
```
Comment (alexander-fenster) [inline src/lib/asyncPrIterator.ts:209]: should we also do `processIssues ? "issue" : "PR"` here?

**Comment by bcoe** [comment]
> One question though, why does your sample output show Unable to process (even though your processMethod always returns true)?

I believe so? I was just copying the `list` command and switching it to issues (which also returns `true`).

**Pull request merged**